src/zope/interface/registry.py

```diff
@@ -184,6 +145,9 @@ def __init__(self, name='', bases=()):
         self._init_registrations()
         self.__bases__ = tuple(bases)
 
+        if hasattr(self, '_v_utility_registrations_cache'):
```

Comment:
`Components` is not a subclass of persistent so there's no guarantee that `_v` attributes will *not* be written out in pickles. That seems...suboptimal? I know that there are other subclasses of `Components` besides the ones defined in `zope.component` (for example, pyramid has `class Registry(Components, dict)`, so just overriding `__getstate__` to return `self.__dict__` with `_v_whatever` popped won't work).

I was strongly motivated to keep that from happening in the original PR, although I can't really recall why at this moment. Perhaps this isn't a problem in practice, but it seems worth mentioning.